crates/core/tedge_write/tests/main.rs

```diff
@@ -0,0 +1,84 @@
+use std::fs;
+use std::fs::Permissions;
+use std::os::unix::fs::PermissionsExt;
+use std::path::PathBuf;
+
+use assert_cmd::Command;
+use tempfile::TempDir;
+
+#[test]
+fn creates_dest_file_if_doesnt_exist() {
+    // Arrange
+    let (temp_dir, source_path) = setup_source_file();
+    let destination_path = temp_dir.path().join("destination.txt");
+
+    let mut command = Command::cargo_bin(env!("CARGO_PKG_NAME")).unwrap();
+    command.pipe_stdin(&source_path).unwrap();
+    command.arg(&destination_path);
+
+    // Act
+    command.assert().success();
+
+    // Assert
+    assert_eq!(
+        fs::read_to_string(source_path).unwrap(),
+        fs::read_to_string(destination_path).unwrap(),
+    );
+}
+
+#[test]
+fn changes_file_permissions_if_file_doesnt_exist() {
```

Comment:
If the file doesn't exist, the permissions can't be changed as they don't exist, so it feels like this name needs tweaking.